# ImageDDS

# Abstract
In synergistic drug combination therapy, computational screening has become a crucial tool for identifying optimal drug combinations. In recent years, graph neural networks (GNNs) have emerged as the primary method for extracting drug features in synergy prediction. However, conventional GNN models still face limitations in capturing complex chemical structures, particularly for cyclic molecules and structures with a large number of atoms. In this work, we propose a multi-modal deep learning framework, ImageDDS, that integrates molecular graphs and images, and leverages self-supervised contrastive learning to align chemical knowledge across both graphs and images for drug synergy prediction. Specifically, we employ a graph Transformer and ResNet18 to extract features from molecular graphs and images, respectively, and fuse these modality-specific features to create a global drug representation. During this process, we further apply contrastive learning to optimize the alignment of graph and image representations, ensuring consistency of features for the same drug across different modalities. Finally, we combine the multi-modal drug features with gene expression embedding from cell lines and feed them into a predictor to identify synergistic drug combinations. In benchmark datasets, we compare ImageDDS with classical machine learning methods and other deep learning-based approaches, with leave-one-out cross-validation demonstrating the superior performance of ImageDDS. Furthermore, we explore the interpretability of graph and image models in discovering important substructures, revealing the crucial role of image features in identifying key drug substructures, especially cyclic structures.

![model]([ImageDDS.jpg](https://github.com/AnQi-87/ImageDDS/blob/main/ImageDDS.jpg))

# Environment
